### PR TITLE
Update freetype to 2.6.5

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/libfreetype2/21)

### CODE REVIEW

**Comment by nox** [comment]
This cannot be reviewed on GitHub because I properly imported upstream's history and used `git subtree`.

@vvuk @metajack Are my changes to build.rs ok? Specifically the part where I changed the minimal version that `pkg-config` searches for.

**Comment by nox** [comment]
@metajack There were no Android-specific fixes in the sources of freetype itself btw.

**Comment by metajack** [comment]
Did you check what the fixes were to the old code? Do they seem important? Or are we going to scramble to fix those on Android when this lands?

**Comment by nox** [comment]
@metajack I didn't find any fix in the old source code, only in our own stuff around it. Which one do you have in mind?

**Comment by metajack** [comment]
Can you post a diff? It's been ages since I've looked at this.

**Comment by nox** [comment]
https://gist.github.com/nox/d28747fe6ae9dfe0e749395c797e3ca3

And the diffstat:

```
 .gitignore                                 |     4 +-
 .mailmap                                   |     7 -
 .travis.yml                                |     7 +
 CMakeLists.txt                             |   357 +
 Cargo.toml                                 |    18 +
 build.rs                                   |    21 +
 builds/unix/.gitignore                     |    17 -
 builds/unix/aclocal.m4                     |  8601 +++++++++++++++
 builds/unix/config.guess                   |  1537 +++
 builds/unix/config.sub                     |  1789 ++++
 builds/unix/configure                      | 15038 +++++++++++++++++++++++++++
 builds/unix/configure.ac                   |   740 ++
 builds/unix/install-sh                     |   527 +
 builds/unix/ltmain.sh                      |  9655 +++++++++++++++++
 builds/unix/unix-cc.in                     |     2 +-
 builds/win32/.gitignore                    |     5 -
 docs/reference/.gitignore                  |     1 -
 docs/reference/ft2-auto_hinter.html        |   313 +
 docs/reference/ft2-base_interface.html     |  3616 +++++++
 docs/reference/ft2-basic_types.html        |  1173 +++
 docs/reference/ft2-bdf_fonts.html          |   260 +
 docs/reference/ft2-bitmap_handling.html    |   302 +
 docs/reference/ft2-bzip2.html              |    94 +
 docs/reference/ft2-cache_subsystem.html    |  1170 +++
 docs/reference/ft2-cid_fonts.html          |   204 +
 docs/reference/ft2-computations.html       |   832 ++
 docs/reference/ft2-font_formats.html       |    84 +
 docs/reference/ft2-gasp_table.html         |   142 +
 docs/reference/ft2-glyph_management.html   |   673 ++
 docs/reference/ft2-glyph_stroker.html      |   940 ++
 docs/reference/ft2-glyph_variants.html     |   269 +
 docs/reference/ft2-gx_validation.html      |   356 +
 docs/reference/ft2-gzip.html               |    94 +
 docs/reference/ft2-header_file_macros.html |   869 ++
 docs/reference/ft2-incremental.html        |   401 +
 docs/reference/ft2-index.html              |   298 +
 docs/reference/ft2-lcd_filtering.html      |   192 +
 docs/reference/ft2-list_processing.html    |   486 +
 docs/reference/ft2-lzw.html                |    94 +
 docs/reference/ft2-mac_specific.html       |   368 +
 docs/reference/ft2-module_management.html  |   818 ++
 docs/reference/ft2-multiple_masters.html   |   511 +
 docs/reference/ft2-ot_validation.html      |   208 +
 docs/reference/ft2-outline_processing.html |  1147 ++
 docs/reference/ft2-pfr_fonts.html          |   206 +
 docs/reference/ft2-quick_advance.html      |   185 +
 docs/reference/ft2-raster.html             |   606 ++
 docs/reference/ft2-sfnt_names.html         |   232 +
 docs/reference/ft2-sizes_management.html   |   164 +
 docs/reference/ft2-system_interface.html   |   415 +
 docs/reference/ft2-toc.html                |   227 +
 docs/reference/ft2-truetype_engine.html    |   132 +
 docs/reference/ft2-truetype_tables.html    |  1232 +++
 docs/reference/ft2-type1_tables.html       |   689 ++
 docs/reference/ft2-user_allocation.html    |    47 +
 docs/reference/ft2-version.html            |   219 +
 docs/reference/ft2-winfnt_fonts.html       |   278 +
 lib.rs                                     |     1 +
 objs/.gitignore                            |    14 -
 src/tools/docmaker/.gitignore              |     1 -
 60 files changed, 58841 insertions(+), 47 deletions(-)
```

**Comment by metajack** [comment]
I don't understand your diff. If you look at the commit history the second commit is a change to unix-cc.in. That is nowhere in your diff.

There were some changes in very first commit relative to the official version which they started from. Those are the changes I'm wondering about. All the work since then has been cargoization, which obviouslly isn't affected much by upgrading freetype from upstream.

**Comment by metajack** [comment]
Sorry. GitHub was cutting off the diff content. Rechecking.

**Comment by nox** [comment]
Here is the diffstat between upstream 2.4.11 tree and first commit in the master branch:

```
± git diff --stat VER-2-4-11 2266610ffeb1d355ebc763b33115e08191b1d850
 .gitignore                                 |     1 -
 .mailmap                                   |     7 -
 builds/unix/.gitignore                     |    17 -
 builds/unix/aclocal.m4                     |  8601 +++++++++++++++++++++++++++++++++++++
 builds/unix/config.guess                   |  1537 +++++++
 builds/unix/config.sub                     |  1789 ++++++++
 builds/unix/configure                      | 15038 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 builds/unix/configure.ac                   |   740 ++++
 builds/unix/install-sh                     |   527 +++
 builds/unix/ltmain.sh                      |  9655 ++++++++++++++++++++++++++++++++++++++++++
 builds/win32/.gitignore                    |     5 -
 docs/reference/.gitignore                  |     1 -
 docs/reference/ft2-auto_hinter.html        |   313 ++
 docs/reference/ft2-base_interface.html     |  3616 ++++++++++++++++
 docs/reference/ft2-basic_types.html        |  1173 +++++
 docs/reference/ft2-bdf_fonts.html          |   260 ++
 docs/reference/ft2-bitmap_handling.html    |   302 ++
 docs/reference/ft2-bzip2.html              |    94 +
 docs/reference/ft2-cache_subsystem.html    |  1170 +++++
 docs/reference/ft2-cid_fonts.html          |   204 +
 docs/reference/ft2-computations.html       |   832 ++++
 docs/reference/ft2-font_formats.html       |    84 +
 docs/reference/ft2-gasp_table.html         |   142 +
 docs/reference/ft2-glyph_management.html   |   673 +++
 docs/reference/ft2-glyph_stroker.html      |   940 +++++
 docs/reference/ft2-glyph_variants.html     |   269 ++
 docs/reference/ft2-gx_validation.html      |   356 ++
 docs/reference/ft2-gzip.html               |    94 +
 docs/reference/ft2-header_file_macros.html |   869 ++++
 docs/reference/ft2-incremental.html        |   401 ++
 docs/reference/ft2-index.html              |   298 ++
 docs/reference/ft2-lcd_filtering.html      |   192 +
 docs/reference/ft2-list_processing.html    |   486 +++
 docs/reference/ft2-lzw.html                |    94 +
 docs/reference/ft2-mac_specific.html       |   368 ++
 docs/reference/ft2-module_management.html  |   818 ++++
 docs/reference/ft2-multiple_masters.html   |   511 +++
 docs/reference/ft2-ot_validation.html      |   208 +
 docs/reference/ft2-outline_processing.html |  1147 +++++
 docs/reference/ft2-pfr_fonts.html          |   206 +
 docs/reference/ft2-quick_advance.html      |   185 +
 docs/reference/ft2-raster.html             |   606 +++
 docs/reference/ft2-sfnt_names.html         |   232 +
 docs/reference/ft2-sizes_management.html   |   164 +
 docs/reference/ft2-system_interface.html   |   415 ++
 docs/reference/ft2-toc.html                |   227 +
 docs/reference/ft2-truetype_engine.html    |   132 +
 docs/reference/ft2-truetype_tables.html    |  1232 ++++++
 docs/reference/ft2-type1_tables.html       |   689 +++
 docs/reference/ft2-user_allocation.html    |    47 +
 docs/reference/ft2-version.html            |   219 +
 docs/reference/ft2-winfnt_fonts.html       |   278 ++
 objs/.gitignore                            |    14 -
 src/tools/docmaker/.gitignore              |     1 -
 54 files changed, 58433 insertions(+), 46 deletions(-)
```

The change in unix-cc.in is to remove the version number from the lib artifact, but AFAIK this only matters for shared libs.

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 language: rust
-sudo: false
```
Comment (metajack) [inline .travis.yml]: Can we not keep using sudo: false? Doesn't that put us on the slow builders?

**Pull request merged**